File: backend/mr/src/main/java/io/featurehub/mr/resources/auth/BlankProvider.java
package io.featurehub.mr.resources.auth;

import javax.ws.rs.NotFoundException;
import java.util.Collection;
import java.util.Collections;

// ensures we have at least one of these as injection requires it
public class BlankProvider implements AuthProvider {
  @Override
  public Collection<String> getProviders() {
    return Collections.emptyList();
  }

  @Override
  public String requestRedirectUrl(String provider) {
    // if we get picked there is something seriously wrong
    throw new NotFoundException();
  }
}

File: backend/security-oauth/src/main/kotlin/io/featurehub/web/security/oauth/BlankProvider.kt
package io.featurehub.web.security.oauth

import javax.ws.rs.NotFoundException

// ensures we have at least one of these as injection requires it
class BlankProvider : AuthProvider {
    override val providers: Collection<String>
        get() = emptyList<String>()

    override fun requestRedirectUrl(provider: String): String {
        // if we get picked there is something seriously wrong
        throw NotFoundException()
    }
}
